python/ray/serve/tests/test_serve_ha.py

```diff
@@ -1,61 +1,38 @@
+import os
 import pytest
 import sys
 import threading
 from time import sleep
+
 from ray.tests.conftest_docker import *  # noqa
 from ray._private.test_utils import wait_for_condition
 from ray._private.resource_spec import HEAD_NODE_RESOURCE_NAME
 
 scripts = """
-import ray
 import json
-from fastapi import FastAPI
-app = FastAPI()
+import os
+
+import ray
+
 from ray import serve
-ray.init(address="auto", namespace="g")
 
-@serve.deployment(num_replicas={num_replicas})
+@serve.deployment
 @serve.ingress(app)
-class Counter:
-    def __init__(self):
-        self.count = 0
-
-    @app.get("/")
-    def get(self):
-        return {{"count": self.count}}
-
-    @app.get("/incr")
-    def incr(self):
-        self.count += 1
-        return {{"count": self.count}}
-
-    @app.get("/decr")
-    def decr(self):
-        self.count -= 1
-        return {{"count": self.count}}
-
-    @app.get("/pid")
-    def pid(self):
-        import os
+class GetPID:
+    def __call__(self, *args):
         return {{"pid": os.getpid()}}
 
-counter_app = Counter.bind()
-serve.run(target=counter_app, name="Counter", route_prefix="/api")
+serve.run(GetPID.options(num_replicas={num_replicas}).bind())
 """
 
 check_script = """
+import ray
 import requests
-import json
-if {num_replicas} == 1:
-    b = json.loads(requests.get("http://127.0.0.1:8000/api/").text)["count"]
```

Comment:
this was fully redundant with checking the number of PIDs

Reply:
checking the PID doesn't change has the same effect (assuming we trust python class implementation :))